### ISSUE TITLE
Only install @babel packages if certain conditions are met

### ISSUE THREAD

**Issue by devinrhode2** [jeffijoe/typesync#80]
I did some analysis on what the `@types/babel__core` and `@types/babel__preset-env` packages actually are.

If someone does not even have a babel config file in their project - installing these is definitely a waste of disk and network resources

I'm not sure if typescript will do anything with `.babelrc` (json) files, but certainly, there's a case for installing these `@types/babel__FOO` packages, if someone has some sort of `babel.config.js` file, I've seen `next.config.js` files have types applied like so: `/** @type {import('next').NextConfig} */`

<img width="570" alt="CleanShot 2021-11-16 at 22 33 21@2x" src="https://user-images.githubusercontent.com/539816/142129943-ca2fce86-2a7e-4cca-9faa-1bf2e0e79e23.png">

Presumably, same thing could be done with babel configs:
```
/** @type {import('@types/babel__core').MainConfigFileType} */
```

So, I don't now all the conditions in which @babel types should be installed, but these are maybe the two ends of the spectrum.

Open questions:
1. Can typescript provide type checking for `.babelrc`? Do we even care about this?
2. What are all the valid ways to define a _javascript_ babel config file?
<sub>Since we can have type checking on javascript files, I assume we should not concern ourselves with supporting non-standard `.ts` config files</sub>
- Should we try and use cosmic-config to find all the babel js config files?

====

I think given the simple fact that the babel types packages only help type-check babel config files, and I'm guessing most people are not really worried very much about that, maybe we special-case @babel packages, until we have something more sophisticated that makes these packages actually useful?
- One thing we could do:
 1. Normalize babel config to a `babel.config.js` (or maybe `.babelrc.js`) file
 2. Add `/** @type {import('@types/babel__core').MainConfigFileType} */` right above the `module.exports` in that file.
We could have an alternate approach for ESM - but I think it'd be better to normalize to either cjs or esm - and I think cjs is the safer+easier choice here. 

TL;DR: installing @babel @types is currently a waste

**Comment by devinrhode2** [comment]
Ok this is even crazier than I was expecting...
<img width="663" alt="CleanShot 2021-11-16 at 22 59 23@2x" src="https://user-images.githubusercontent.com/539816/142132199-ad121d2e-a90f-43d8-ab50-3c60032d6acf.png">

typescript folks need to take a break

**Comment by jeffijoe** [comment]
I think a simpler solution is to provide a starting typesync config file that ignores a few high profile tools like babel and eslint.

**Comment by devinrhode2** [comment]
I think whichever solution we go with, we'll at least want a little blurb in the readme mentioning e.g. how to type check eslintrc.js

**Comment by jeffijoe** [comment]
I didn’t even realize one could do that this way 😄

A blur in the readme for ignoring tool packages makes sense.

**Comment by devinrhode2** [comment]
Well I think everyone is going to be curious about these types.. so "yes you can ignore them, but if you want you can also use them like so;"

**Comment by devinrhode2** [comment]
I realized there's another dimension here.

We can use this `/** @type {FOO} */` syntax to add types in javascript files. But.. I've found it's a little tricky to configure tsconfig.json to properly check these `.*rc.js` files. Ideally developers see issues inside vscode, but we want a `yarn tsc` or `yarn custom-tsc-command` to inform us of issues before code is merged into main.

It's hard to say if `.ts` config files are the future. (related: https://github.com/vercel/next.js/issues/5318) There's even a "kernel" of deno that is even in plain javascript. I'd expect the `/** @type {FOO} */` syntax could be quite popular for 1-4 years, and possibly stay around for 15-20 years, when/if typescript passes the torch to a "native" javascript type system.
